app/src/main/java/com/example/development/sakaiclientandroid/utils/DataHandler.java

```diff
@@ -185,14 +197,14 @@ public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response)
 
                 hasRequestedAllGrades = false;
                 hasRequestedAllAssignments = false;
-                UICallback.onCoursesSuccess(coursesSortedByTerm);
+                UICallback.onAllCoursesSuccess(coursesSortedByTerm);
```

Comment:
Same thing here. Call `UICallback.onAllCourseFailure()` in the `catch` block and return if the parsing fails, otherwise the parsing may fail and we might still end up calling `onAllCoursesSuccess`